src/video/SDL_surface.c

```diff
@@ -195,6 +205,22 @@ SDL_CreateRGBSurfaceFrom(void *pixels,
 {
     SDL_Surface *surface;
 
+    if (width < 0) {
+        SDL_InvalidParamError("width");
+        return NULL;
+    }
+
+    if (height < 0) {
+        SDL_InvalidParamError("height");
+        return NULL;
+    }
+
+    /* pitch >= (width * bytes per pixel) >= width >= 0 */
+    if (pitch < width) {
```

Comment:
Does this account for bitmap formats where pitch is in bytes but width is in bits?

Reply:
Probably not. Would that be the formats that use `SDL_PIXELTYPE_INDEX1` and `SDL_PIXELTYPE_INDEX4`? Do those formats pack more than one pixel into each byte?

> width is in bits

I'm confused by this: surely the width is always measured in pixels, regardless of how many bits a pixel might occupy?

Do you mean width is in pixels, with fewer than 8 bits in a pixel and therefore potentially more than one pixel in a byte?

Reply:
Yes, exactly that. `SDL_PIXELTYPE_INDEX1` and `SDL_PIXELTYPE_INDEX4` pack more than one pixel per byte, so you could have an 8x8 `SDL_PIXELTYPE_INDEX1` surface that has width 8 and pitch 1.

Reply:
OK, this is a bit more complicated but I think it should be correct now.

If we want to check that the pitch parameter is valid, we now need to be able to calculate two values for the pitch: the optimized (rounded-up) value that SDL will use if left to its own devices, and the minimal value that is the smallest pitch SDL can possibly accept from third-party code (just enough bytes to hold a scanline).

Or, I could drop the last two commits to just have minimal validation that the pitch/rowstride is non-negative (and if people pass in 0 and all their rows are occupying the same memory, that's on them).

Reply:
I think non-negative is a sufficient check, but I accepted your patch as-is.

Reply:
Oh, never mind, you merged the version before I split those commits. That's functionally equivalent.